frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -55,13 +86,50 @@ export function DynamicConfiguration () {
         formSubmitting: false as boolean,
     });
 
+    let formErrors = {} as any;
+    const parseResult = formSchema.safeParse(state.form);
+    if (!parseResult.success) {
+        formErrors = issuesToErrors(
+            parseResult.error.issues,
+            state.validateAll === true ? undefined : state.touched
+        );
+    }
+
     useEffect(() => {
         dispatch(listConfigurations({configScope: 'global', numVersions: 5}));
     }, [dispatch]);
 
+    useEffect(() => {
+        dispatch(listEMRApplications());
+    }, [dispatch]);
+
+    useEffect(() => {
+        const appList: any[] = [];
+        for (const application of applicationList) {
+            const app = {
+                value: application,
+                label: application
+            };
+            appList.push(app);
+        }
+        // Make sure we're always starting from an empty array to prevent duplicates
+        setSelectedApplicationOptions([]);
+        for (const configApp of applicationConfig.configuration.EMRConfig.applications) {
+            if (applicationList.includes(configApp.Name) && !selectedApplicationOptions.includes(configApp.Name)) {
+                const app = {
+                    value: configApp.Name,
+                    label: configApp.Name
+                };
+                setSelectedApplicationOptions((priorSelectedOptions) => [...priorSelectedOptions, app]);
+            }
+        }
+        setApplicationOptions(appList);
```

Comment:
Since this isn't modified after line 113, I would move it up to that codeblock for better co-location of code.